src/components/common/validation/time-format/ValidatedTimeEntry.jsx

```diff
@@ -1,60 +1,23 @@
 import PropTypes from 'prop-types';
-import { useTimecardContext } from '../../../../context/TimecardContext';
-import { deepCloneJson } from '../../../../utils/common-utils/common-utils';
-import { inputNames } from '../../../../utils/constants';
-import { ContextTimeEntry } from '../../../../utils/time-entry-utils/ContextTimeEntry';
-import { isFinishTimeOnNextDay } from '../../../../utils/time-entry-utils/timeEntryUtils';
+import { useEffect } from 'react';
 
 const ValidatedTimeEntry = ({
   name,
-  timeType,
   errors,
   defaultValue,
   register,
-  isRequired,
-  getFormValues,
-  timeEntry,
-  timeEntriesIndex,
+  updateFinishTimeText,
 }) => {
-  const { timeEntries, setTimeEntries } = useTimecardContext();
-
-  const setFinishTimeText = () => {
-    const startTimeValue = getFormValues(inputNames.shiftStartTime);
-    const finishTimeValue = getFormValues(inputNames.shiftFinishTime);
-    const newTimeEntries = deepCloneJson(timeEntries);
-
-    const answer = isFinishTimeOnNextDay(startTimeValue, finishTimeValue);
-
-    newTimeEntries[timeEntriesIndex] =
-      ContextTimeEntry.createFrom(timeEntry).setFinishNextDay(answer);
-    setTimeEntries(newTimeEntries);
-  };
-
-  const isTimeValid = (time) => {
-    if (time.length < 3 && time.length > 0) {
-      const hhTimeRegEx = /^(\d|[01]\d|2[0-3])$/;
-      return hhTimeRegEx.test(time);
-    } else if (time.length > 3 && time.length < 6) {
-      const hhmmTimeRegEx = /^([01]\d|2[0-3]):?([0-5]\d)$/;
-      return hhmmTimeRegEx.test(time);
-    } else if (time.length == 0 && timeType === 'finish time') {
-      return true;
-    } else {
-      return false;
-    }
-  };
-
-  const errorMessage = `Enter a ${timeType} in the 24 hour clock format, for example, 08:00 or 0800`;
-
+  useEffect(() => {
+    updateFinishTimeText();
+  }, []);
   return (
     <input
       id={name}
       name={name}
       className={`govuk-input ${
         Object.keys(errors).length !== 0 &&
-        Object.keys(errors).find((error) => {
-          return error === name;
-        })
+        Object.values(errors).some((error) => error.inputName === name)
```

Comment:
errors is now an array so we shouldn't use Object here

Reply:
Removed all Object references in reation to errors